homeassistant/components/dremel_3d_printer/__init__.py

```diff
@@ -0,0 +1,155 @@
+"""The Dremel 3D Printer (3D20, 3D40, 3D45) integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from dremel3dpy import Dremel3DPrinter
+from requests.exceptions import ConnectTimeout, HTTPError
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+import homeassistant.util.dt as dt_util
+
+from .const import _LOGGER, DOMAIN, PLATFORMS, SCAN_INTERVAL
+from .services import async_setup_services
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the OctoPrint component."""
+    if DOMAIN not in config:
+        return True
+
+    domain_config = config[DOMAIN]
+
+    for conf in domain_config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    CONF_HOST: conf[CONF_HOST],
+                },
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Dremel 3D Printer from a config entry."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    try:
+        api = await hass.async_add_executor_job(
+            Dremel3DPrinter, config_entry.data[CONF_HOST]
+        )
+
+    except (ConnectTimeout, HTTPError) as exc:
+        raise ConfigEntryNotReady(
+            f"Unable to connect to Dremel 3D Printer: {exc}"
+        ) from exc
+    except Exception as exc:
+        raise ConfigEntryNotReady(
+            f"Unknown error connecting to Dremel 3D Printer: {exc}"
+        ) from exc
+
+    coordinator = Dremel3DPrinterDataUpdateCoordinator(
+        hass, f"{DOMAIN}-{config_entry.entry_id}", api, SCAN_INTERVAL.seconds
+    )
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+    await async_setup_services(hass)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Dremel config entry."""
+
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        del hass.data[DOMAIN]
```

Comment:
```suggestion
    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        del hass.data[DOMAIN]
```

Reply:
> Contributor

It would be awesome if you could test it on your Dremel 3D Printer. Even more awesome if you hold the versions 3D20 or 3D40, which don't have camera, to see if the API works the same way as it does on the 3D45 model and it adds the same entities as it does for the 3D45 model except the Camera. Otherwise, having another pair of eyes testing it on the 3D45 model would be appreciated as well :-)

I've added the test coverage to `config_flow` and will be changing this diff to only include the `Sensor` platform.

By the way, if you don't mind me asking, I've looked into the documentation but couldn't find directions on how to align this PR with the changes to 1. add documentation to Home Assistant docs and 2. add a logo for this integration on the Integrations UI. Can you link me to the documentation on how to do so or simply guide me on how can I make progress on that? I understand there are two different repos for docs and for logo respectively, but I'm lost as to how can I align PRs made to those repos with this PR, so that they are published together, or in other words, what's the correct order of repos that I should make changes to (adding docs first, then adding logo, then publishing this diff? Publishing this diff first? etc.)

Reply:
Also, after this PR gets accepted, should I keep adding the other platforms to this same PR or open a new one?